app/models/audit.rb

```diff
@@ -0,0 +1,99 @@
+require 'syslog'
+require 'logger'
+require 'time'
+
+module Audit
+  class << self
+    def notice msg, msgid, data
+      logger.info LogMessage.new msg, msgid, data, Syslog::LOG_NOTICE
+    end
+
+    def logger
+      @logger ||= Rails.logger
+    end
+
+    attr_writer :logger
+  end
+
+  module SDID
+    CONJUR_PEN = 43838
+    def self.conjur_sdid label
+      [label, CONJUR_PEN].join('@').freeze
+    end
+
+    POLICY = conjur_sdid 'policy'
+    AUTH = conjur_sdid 'auth'
+    SUBJECT = conjur_sdid 'subject'
+    ACTION = conjur_sdid 'action'
+  end
+
+  class LogMessage < String
+    def initialize msg, msgid, structured_data = nil, severity = nil
+      super msg
+      @msgid = msgid
+      @structured_data = structured_data
+      @severity = severity
+    end
+
+    attr_reader :msgid, :structured_data, :severity
+  end
+
+  # Middleware to store request ID in a thread variable
+  class RememberUuid < Struct.new :app
+    def call env
+      Thread.current[:request_id] = env["action_dispatch.request_id"]
```

Comment:
Maybe `conjur_request_id` for safety

Reply:
I kept going back and forth on this. However there's nothing Conjur-specific about this request ID.